src/faebryk/exporters/esphome/esphome.py

```diff
@@ -53,9 +53,7 @@ def merge_dicts(*dicts: dict) -> dict:
 
 
 def make_esphome_config(G: Graph) -> dict:
-    from faebryk.core.util import get_all_nodes_with_trait
-
-    esphome_components = get_all_nodes_with_trait(G, F.has_esphome_config)
+    esphome_components = G.nodes_with_trait(F.has_esphome_config)
```

Comment:
Is the implication of using a `Graph`, rather than a `Node` that you can pre-process it with other filters etc...

Reply:
Yeah, but also node should be only used if we respect the hierarchy.
Since the esphome exporter looks for connected buses this breaks hierarchy and thus is better of using the full graph.